src/Microsoft.AspNet.Mvc.TagHelpers/EnvironmentTagHelper.cs

```diff
@@ -67,25 +67,33 @@ public override void Process(TagHelperContext context, TagHelperOutput output)
                 return;
             }
 
-            var environments = Names.Split(NameSeparator, StringSplitOptions.RemoveEmptyEntries)
-                                    .Where(name => !string.IsNullOrWhiteSpace(name));
-
-            if (!environments.Any())
+            var currentEnvironmentName = HostingEnvironment.EnvironmentName?.Trim();
+            if (string.IsNullOrWhiteSpace(currentEnvironmentName))
             {
-                // Names contains only commas or empty entries, do nothing
+                // No current environment name, do nothing
                 return;
             }
 
-            var currentEnvironmentName = HostingEnvironment.EnvironmentName?.Trim();
+            var values = Names.Split(NameSeparator, StringSplitOptions.RemoveEmptyEntries);
+            var environments = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
 
-            if (string.IsNullOrWhiteSpace(currentEnvironmentName))
+            for (var i = 0; i < values.Length; i++)
             {
-                // No current environment name, do nothing
+                var trimmedValue = values[i].Trim();
+
+                if (trimmedValue.Length > 0)
+                {
+                    environments.Add(trimmedValue);
+                }
+            }
+
+            if (environments.Count == 0)
+            {
+                // Names contains only commas or empty entries, do nothing
                 return;
             }
 
-            if (environments.Any(name =>
-                string.Equals(name.Trim(), currentEnvironmentName, StringComparison.OrdinalIgnoreCase)))
+            if (environments.Contains(currentEnvironmentName))
```

Comment:
This could be much easier done with a tokenizer.

basically `string.IndexOf(currentEnvironmentName)` and then check the surrounding characters for whitespace/comma